src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -657,9 +657,7 @@ export default function Swap({
                     // gap: 8 // mod
                   }}
                 >
-                  {trade && (
-                    <Price trade={trade} theme={theme} showInverted={showInverted} setShowInverted={setShowInverted} />
-                  )}
+                  <Price trade={trade} theme={theme} showInverted={showInverted} setShowInverted={setShowInverted} />
```

Comment:
But why? I would think the fault of memory leaks would be inside `Price` not in `SwapMod`. What is the issue of conditionally rendering a component? 

If you change it for memory leaks here, u are doing a "white box" fix, where u know the implementation of Price does something bad when unmounted. I understand SwapMod wouldn't need to know anything about the Price component internals (should treat it as a "black box").

Let me know if my reasoning above is incorrect, likely missing sth here.

Reply:
Because price component was rendering a child that was calculating state that when un mounted was not cancelling the asynchronous promise

Reply:
This is what i mean, the problem is in price, not in SwapMod, isn't it?

Reply:
i see it 2 ways:
1. SwapMod has the data sources/biz logic, and Price is dumb. It gets all the props etc, but now it needs a wrapping container to be independent and reusable elsewhere - SwapMod is the smart container in this example and relies on it to render it it's smart props
2. it has a closer related wrapping container that isn't SwapMod that handles the biz logic and can be used as 1 component in other places (which is what i assume is what u intended with the refactor) - so now inside `/custom/swap/index.ts` we have the logic inside the parent containers that allow the entire components to be reusable. for Price you can see that logic here: https://github.com/gnosis/cowswap/pull/1365/files#diff-e1d5a26ef1f0d5d518088887aa8e77620e509679ab3a6138b94360dc722102ccR196
which we can export and use everywhere and it will always render the price in USD

does that make sense?